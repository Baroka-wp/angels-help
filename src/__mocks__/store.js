const store = [
  {
    order: 1,
    hierarchy: 'Seraphim',
    fullName: 'VEHUIAH',
    definitionOfName: 'God exalted and exalted above all things',
    godName: 'JEHOVAH',
    description: 'N/A',
    regenceMonth: 'March',
    regenceDays: '21-25',
    regenceHours: 'N/A',
    colors: 'purple',
    scent: 'Electre',
    gender: 'Femal',
    invocation: 'Make your powers take root in me. Amen',
    exhortation: 'The time.',
    callFormula: 'But you, o VEHUIAH, You are my honor and hold my head high',
    psalms: 'Ps3:4',
    category: 'health',
  },
  {
    order: 6,
    hierarchy: 'Seraphim',
    fullName: 'LELAHEL',
    definitionOfName: 'Praiseworthy god',
    godName: 'ABGED',
    description: 'N/A',
    regenceMonth: 'April',
    regenceDays: '15-20',
    regenceHours: 'N/A',
    colors: 'purple',
    scent: 'Santal',
    gender: 'Male',
    invocation: 'Lelahel! I give you thanks, Lord, for this stopover,  Amen!',
    exhortation: 'I gave you powers to create, so that eternal truths ',
    callFormula: 'Celebrate the Lord who sits in Zion; proclaim among the peoples his deeds',
    psalms: 'Ps9:12',
    category: 'Love',
  },
];

export default store;
