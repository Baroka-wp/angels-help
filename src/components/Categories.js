import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { FiArrowRightCircle } from 'react-icons/fi';
import {
  GiReceiveMoney,
  GiRoyalLove,
  GiOfficeChair,
  GiHealing,
  GiSecurityGate,
  GiOlive,
} from 'react-icons/gi';
import { fetchAgels, filterAngels } from '../redux/angels/angels';

const Categories = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchAgels());
  }, []);
  const state = useSelector((state) => state.angels);
  const countAngelsbyCategory = (c) => {
    const temp = [...state].filter((item) => item.category === c);
    return temp.length;
  };
  const filterby = (category) => {
    dispatch(filterAngels(category));
  };

  return (
    <>
      <Link
        to="/angels-help/angelsList"
        name="money"
        onClick={() => filterby('Money')}
        className="card"
      >
        <div className="sumary">
          <div className="icone">
            <GiReceiveMoney className="card-icone" />
          </div>
          <div className="details">
            <div>
              <h3>Money</h3>
              <p>
                {countAngelsbyCategory('Money')}
                &nbsp;Angels
              </p>
            </div>
            <FiArrowRightCircle />
          </div>
        </div>
      </Link>
      <Link
        to="/angels-help/angelsList"
        onClick={() => filterby('Love')}
        className="card ligth-card"
      >
        <div className="sumary">
          <div className="icone">
            <GiRoyalLove className="card-icone" />
          </div>
          <div className="details">
            <div>
              <h3>Love</h3>
              <p>
                {countAngelsbyCategory('Love')}
                &nbsp;Angels
              </p>
            </div>
            <FiArrowRightCircle />
          </div>
        </div>
      </Link>
      <Link
        to="/angels-help/angelsList"
        onClick={() => filterby('Job')}
        className="card ligth-card"
      >
        <div className="sumary">
          <div className="icone">
            <GiOfficeChair className="card-icone" />
          </div>
          <div className="details">
            <div>
              <h3>Job</h3>
              <p>
                {countAngelsbyCategory('Job')}
                &nbsp;Angels
              </p>
            </div>
            <FiArrowRightCircle />
          </div>
        </div>
      </Link>
      <Link
        to="/angels-help/angelsList"
        onClick={() => filterby('health')}
        className="card"
      >
        <div className="sumary">
          <div className="icone">
            <GiHealing className="card-icone" />
          </div>
          <div className="details">
            <div>
              <h3>Health</h3>
              <p>
                {countAngelsbyCategory('health')}
                &nbsp;Angels
              </p>
            </div>
            <FiArrowRightCircle />
          </div>
        </div>
      </Link>
      <Link
        to="/angels-help/angelsList"
        onClick={() => filterby('Protection')}
        className="card"
      >
        <div className="sumary">
          <div className="icone">
            <GiSecurityGate className="card-icone" />
          </div>
          <div className="details">
            <div>
              <h3>Protection</h3>
              <p>
                {countAngelsbyCategory('Protection')}
                &nbsp;Angels
              </p>
            </div>
            <FiArrowRightCircle />
          </div>
        </div>
      </Link>
      <Link
        to="/angels-help/angelsList"
        onClick={() => filterby('Spirituality')}
        className="card ligth-card"
      >
        <div className="sumary">
          <div className="icone">
            <GiOlive className="card-icone" />
          </div>
          <div className="details">
            <div>
              <h3>Energy</h3>
              <p>
                {countAngelsbyCategory('Spirituality')}
                &nbsp;Angels
              </p>
            </div>
            <FiArrowRightCircle />
          </div>
        </div>
      </Link>
    </>
  );
};

export default Categories;
