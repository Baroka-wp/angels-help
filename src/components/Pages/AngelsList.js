import React from 'react';
import { FiArrowRightCircle } from 'react-icons/fi';
import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import Spinner from './Spinner';

const AngelsList = () => {
  const angelsList = useSelector((state) => state.angels);
  let categoryName = 'Angel';
  if (angelsList.length > 0) {
    categoryName = [...angelsList][0].category;
  }
  const loading = useSelector((state) => state.loader);

  return (
    <div data-testid="angels-list">
      {loading && <Spinner />}
      <div>
        <div className="hero-cover" />
        <div className="hero">
          <h3>{categoryName}</h3>
          <p>
            Angels are messengers who help you to get in
            touch with God to ask Him for the&nbsp;
            {categoryName}
            &nbsp;whatever you want.
          </p>
        </div>
      </div>
      <div className="angelsListContainer">
        {
          angelsList.map((angel, index) => (
            <Link
              key={angel.order}
              to="/angels-help/angelsDetails"
              state={{ angel }}
              className={index % 2 === 0 ? 'angelsList ligth-card' : 'angelsList'}
            >
              <div className="angel">
                <h3>{angel.fullName}</h3>
                <em>{angel.definitionOfName}</em>
              </div>
              <FiArrowRightCircle />
            </Link>
          ))
        }
      </div>
    </div>
  );
};

export default AngelsList;
