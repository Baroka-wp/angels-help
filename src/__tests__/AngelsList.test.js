import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import AngelsList from '../components/Pages/AngelsList';
import Categories from '../components/Categories';

describe('AngelsList should diplay the six categories', () => {
  test('Renders Category in DOM porperly', async () => {
    render(
      <Provider store={store}>
        <Router>
          <Categories />
        </Router>
      </Provider>,
    );
    const linkButton = screen.getByText(/Money/);
    fireEvent.click(linkButton);
    expect(screen.queryByText(/Money/)).toBeTruthy();
  });
  it('render properly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <AngelsList />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
