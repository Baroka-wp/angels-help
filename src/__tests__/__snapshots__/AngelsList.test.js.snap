// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AngelsList should diplay the six categories render properly 1`] = `
<div
  data-testid="angels-list"
>
  <div
    className="loader"
  />
  <div>
    <div
      className="hero-cover"
    />
    <div
      className="hero"
    >
      <h3>
        Angel
      </h3>
      <p>
        Angels are messengers who help you to get in touch with God to ask Him for the 
        Angel
         whatever you want.
      </p>
    </div>
  </div>
  <div
    className="angelsListContainer"
  />
</div>
`;
